examples/server/server.cpp

```diff
@@ -684,125 +811,173 @@ static void server_params_parse(int argc, char ** argv, server_params & sparams,
+static json format_generation_settings(llama_server_context &llama)
+{
     const auto eos_bias = llama.params.logit_bias.find(llama_token_eos());
     const bool ignore_eos = eos_bias != llama.params.logit_bias.end() &&
-        eos_bias->second < 0.0f && std::isinf(eos_bias->second);
-
-    return json {
-        { "seed", llama.params.seed },
-        { "temp", llama.params.temp },
-        { "top_k", llama.params.top_k },
-        { "top_p", llama.params.top_p },
-        { "tfs_z", llama.params.tfs_z },
-        { "typical_p", llama.params.typical_p },
-        { "repeat_last_n", llama.params.repeat_last_n },
-        { "repeat_penalty", llama.params.repeat_penalty },
-        { "presence_penalty", llama.params.presence_penalty },
-        { "frequency_penalty", llama.params.frequency_penalty },
-        { "mirostat", llama.params.mirostat },
-        { "mirostat_tau", llama.params.mirostat_tau },
-        { "mirostat_eta", llama.params.mirostat_eta },
-        { "penalize_nl", llama.params.penalize_nl },
-        { "stop", llama.params.antiprompt },
-        { "n_predict", llama.params.n_predict },
-        { "n_keep", llama.params.n_keep },
-        { "ignore_eos", ignore_eos },
-        { "stream", llama.stream },
-        { "logit_bias", llama.params.logit_bias },
-        { "n_probs", llama.params.n_probs },
+                            eos_bias->second < 0.0f && std::isinf(eos_bias->second);
+
+    return json{
+        {"n_ctx", llama.params.n_ctx},
+        {"model", llama.params.model_alias},
+        {"seed", llama.params.seed},
+        {"temp", llama.params.temp},
+        {"top_k", llama.params.top_k},
+        {"top_p", llama.params.top_p},
+        {"tfs_z", llama.params.tfs_z},
+        {"typical_p", llama.params.typical_p},
+        {"repeat_last_n", llama.params.repeat_last_n},
+        {"repeat_penalty", llama.params.repeat_penalty},
+        {"presence_penalty", llama.params.presence_penalty},
+        {"frequency_penalty", llama.params.frequency_penalty},
+        {"mirostat", llama.params.mirostat},
+        {"mirostat_tau", llama.params.mirostat_tau},
+        {"mirostat_eta", llama.params.mirostat_eta},
+        {"penalize_nl", llama.params.penalize_nl},
+        {"stop", llama.params.antiprompt},
+        {"n_predict", llama.params.n_predict},
+        {"n_keep", llama.params.n_keep},
+        {"ignore_eos", ignore_eos},
+        {"stream", llama.stream},
+        {"logit_bias", llama.params.logit_bias},
+        {"n_probs", llama.params.n_probs},
+    };
+}
+
+static json format_embedding_response(llama_server_context &llama)
+{
+    return json{
+        {"embedding", llama.getEmbedding()},
     };
 }
 
-static json format_embedding_response(llama_server_context & llama) {
-    return json {
-        { "embedding", llama.getEmbedding() },
+static json format_timings(llama_server_context &llama)
+{
+    const auto timings = llama_get_timings(llama.ctx);
+
+    assert(timings.n_eval == llama.num_tokens_predicted);
```

Comment:
This line also crashes my server every time because they are not the same. I am running everything with the defaults, type "Hello Llama" and it crashes at the assert (guanaco-7B.ggmlv3.q5_K_M).

Commenting this line out seems to work just fine for me although I am not sure what unwanted behaviors I might get. seems like if this is not critical, it should be logged as a warning rather than cause an exception.